api/classes/class-lightweight-api.php

```diff
@@ -333,7 +333,7 @@ function ( $item ) {
 	public function get_all_clients_data() {
 		global $wpdb;
 		$events_table_name   = Segmentation::get_events_table_name();
-		$all_client_ids_rows = $wpdb->get_results( "SELECT DISTINCT client_id FROM $events_table_name" ); // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$all_client_ids_rows = $wpdb->get_results( "SELECT DISTINCT client_id,id FROM $events_table_name ORDER BY id DESC LIMIT 1000" ); // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.PreparedSQL.InterpolatedNotPrepared
```

Comment:
I think it's worth adding a comment here, along the lines of "Results limited to 1000 most recent rows for performance reasons."